### ISSUE TITLE
Document `@woocommerce/shared-hocs`/ `wcBlocksSharedHocs`

### ISSUE THREAD

**Issue by tomalec** [woothemes/woocommerce#42584]
I'm working with AutomateWoo, and noticed that since the introduction of Checkout block (1061-gh-woocommerce/automatewoo) it imports (via DEWP) `@woocommerce/shared-hocs`/ `wcBlocksSharedHocs`.

Is this package considered public, and ready for external use? If so does it have the stable API? API that's documented somewhere, so I would know what my extension is actually using.
I tried to [google it](https://www.google.com/search?q=%40woocommerce%2Fshared-hocs&oq=%40woocommerce%2Fshared-hocs&aqs=chrome..69i57j69i58.374j0j4&sourceid=chrome&ie=UTF-8) but all the results are rather unrelated.
There is [no such package on npm](https://www.npmjs.com/search?q=%40woocommerce%2Fshared-hocs).

I tried searching this repo READMEs and code but still without avail.

While doing so, I managed to find [something](https://github.com/woocommerce/woocommerce-blocks/tree/7757b85b2061853c9710c29cd8706d4b3a620749/packages/checkout) about `@woocommerce/blocks-checkout`. Still, the discoverability was pretty harsh, as I found it only after a source code search.

I think it would be good if both (`shared-hocs` and `blocks-checkout`) would be registered in npm, so it would be easy to find them and their docs. Track versions, make linters happy, etc.

Speaking of `shared-hocs`, if it's meant to be used by external extensions, I think it should have at least some documentation, to state the contract on what's there to expect. Otherwise, I'm importing the very internals of another [extension's `assets`](https://github.com/woocommerce/woocommerce-blocks/tree/trunk/assets/js/hocs) as a black box, without any control or insight of what's there, how to use it, what results to expect. Needless to mention control the version, changes, and tests.

If that's not meant to be used externally, then I'd like to know that as well, so I could try to remove it from AutomateWoo code somehow.

### Related issues 
https://github.com/woocommerce/woocommerce-blocks/issues/5044

//cc @senadir @mikejolley @woocommerce/rubik

**Comment by senadir** [comment]
Thank you for starting this, while I consider `@woocommerce/shared-hocs` as external, it's not setup to be like that, and has a higher risk of breaking, to give you more ease of mind, I suggest you try to work around it and remove it, [we only use `withFilteredAttributes`](https://github.com/woocommerce/automatewoo/pull/1061/files#diff-25a04fe534a34ef425e545ab3faa6ddf70d1df094f396d8d2686c858db1b288d), I think this is something you can work around in your `block.js` file by providing sane defaults from `attributes.js`.

`blocks-checkout` is external and is treated as such, it doesn't have an npm package yet and until we have something like that in place, we will treat it always as backward compatible and won't break anything.

For your use case, I assume you're extracting things dependency extraction plugin, in which having an npm won't be useful for you as it won't be used. Linters and autocomplete is a valid use case for npm thou.

`shared-hocs` predates me a bit, and I think some parts of it is used externally but I can't find anything to it.

**Comment by tomalec** [comment]
Thanks for the quick reply :)

> it doesn't have an npm package yet and until we have something like that in place, we will treat it always as backward compatible and won't break anything.

But do you plan to publish it somewhere in the foreseeable future?

> I assume you're extracting things dependency extraction plugin, in which having an npm won't be useful for you as it won't be used. Linters and autocomplete is a valid use case for npm thou.

Yes, I'm using DEWP, but npm or any source to actually import is needed not only for DevX things like linters and autocomplete, but to be able to run unit tests for anything that even indirectly imports it. Not even use, but just import is enough, to contaigously make writing tests impossible.

**Comment by senadir** [comment]
> But do you plan to publish it somewhere in the foreseeable future?

Yes, but we don't have an exact timeline, this was discussed previously and I think it would make sense to revive that discussion to decide on it.

> Yes, I'm using DEWP, but npm or any source to actually import is needed not only for DevX things like linters and autocomplete, but to be able to run unit tests for anything that even indirectly imports it. Not even use, but just import is enough, to contaigously make writing tests impossible.

Totally agree on this.

**Comment by kkoteen** [comment]
@senadir can you confirm there's still a need for this documentation?
